app/src/main/java/de/vectordata/skynet/net/PacketHandler.java

```diff
@@ -96,21 +95,25 @@ private void handlePacket(byte id, byte[] payload, Packet parent) {
         if (packet == null)
             return;
 
-        if (packet instanceof ChannelMessagePacket)
-            ((ChannelMessagePacket) packet).setParent((P0BChannelMessage) parent);
-
         if (payload == null) {
             Log.w(TAG, "Received corrupted packet 0x" + Integer.toHexString(id));
             return;
         }
 
+        if (packet instanceof ChannelMessagePacket) {
+            ChannelMessagePacket message = (ChannelMessagePacket) packet;
+            Flags flags = message.getClass().getAnnotation(Flags.class);
+            if (flags != null)
+                message.messageFlags |= flags.value();
```

Comment:
It only enforces the flags that are enforced by the protocol to avoid parser errors. I can remove this however if its dangerous. WIll fix this.

Reply:
Throwing an exception instead of swallowing this fault would help to find mistakes in the server.

Reply:
That would be nice, I'd have to do some checking so that optional flags that are sent do not trigger the check. Something like `message.messageFlags | flags.value() == message.messageFlags` maybe.